File: trunk_dxlib/Source/Project/CharacterFactory.cpp
#include "Project/CharacterFactory.hpp"

Character CharacterOf( CharaType type )
{
    switch( type )
    {
    case CharaType_Player:
        return Character( 15 , 4 , 0 );
    case CharaType_GreenSlime:
        return Character( 5 , 2 , 50 );
    case CharaType_BigGreenSlime:
        return Character( 10 , 4 , 50 );
    default:
        return Character( 0 , 0 , 0 );
    }
}

File: trunk_doja/Project/Test/src/Princess/CharacterFactory.java

/**
    キャラクターを生成するクラス.
*/
class CharacterFactory
{
    static final int CharaType_BlueSlime = 0 ;
    static final int CharaType_GreenSlime = 1 ;
    static final int CharaType_Furiru = 2 ;
    
    static Character New( int chara_type )
    {
        switch( chara_type )
        {
        case CharaType_BlueSlime:
            return new Character( "Bスライム" , 4 , 2 , 2 );
        case CharaType_GreenSlime:
            return new Character( "Gスライム" , 8 , 3 , 4 );
        case CharaType_Furiru:
            return new Character( "フリル" , 10 , 4 , 0 );
        default:
            return null;
        }
    }
};


